cli/src/main/java/org/jboss/as/cli/impl/aesh/cmd/security/auth/AbstractDisableAuthenticationCommand.java

```diff
@@ -0,0 +1,121 @@
+limitations under the License.
+ */
+package org.jboss.as.cli.impl.aesh.cmd.security.auth;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.jboss.as.cli.impl.aesh.cmd.security.model.AuthFactorySpec;
+import org.aesh.command.Command;
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandException;
+import org.aesh.command.CommandResult;
+import org.aesh.command.option.Option;
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.cli.CommandFormatException;
+import org.jboss.as.cli.impl.aesh.cmd.security.SecurityCommand;
+import static org.jboss.as.cli.impl.aesh.cmd.security.SecurityCommand.OPT_MECHANISM;
+import org.jboss.as.cli.impl.aesh.cmd.security.model.ElytronUtil;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.core.cli.command.DMRCommand;
+import org.wildfly.core.cli.command.aesh.CLICommandInvocation;
+import static org.jboss.as.cli.impl.aesh.cmd.security.SecurityCommand.OPT_NO_RELOAD;
+
+/**
+ * Base class for any command that disables Auth.
+ *
+ * @author jdenise@redhat.com
+ */
+@CommandDefinition(name = "abstract-auth-disable", description = "")
+public abstract class AbstractDisableAuthenticationCommand implements Command<CLICommandInvocation>, DMRCommand {
+    @Option(name = OPT_MECHANISM,
+            completer = SecurityCommand.OptionCompleters.MechanismDisableCompleter.class)
+    String mechanism;
+
+    @Option(name = OPT_NO_RELOAD, hasValue = false)
+    boolean noReload;
+
+    private final AuthFactorySpec factorySpec;
+
+    public AbstractDisableAuthenticationCommand(AuthFactorySpec factorySpec) {
+        this.factorySpec = factorySpec;
+    }
+
+    public AuthFactorySpec getFactorySpec() {
+        return factorySpec;
+    }
+
+    public abstract String getEnabledFactory(CommandContext ctx) throws Exception;
+
+    protected abstract ModelNode disableFactory(CommandContext context) throws Exception;
+
+    protected abstract String getSecuredEndpoint(CommandContext ctx);
+
+    @Override
+    public CommandResult execute(CLICommandInvocation commandInvocation) throws CommandException, InterruptedException {
+        CommandContext ctx = commandInvocation.getCommandContext();
+        ModelNode request;
+        try {
+            request = buildSecurityRequest(ctx);
+        } catch (Exception ex) {
+            throw new CommandException(ex.getLocalizedMessage());
+        }
+
+        SecurityCommand.execute(ctx, request, SecurityCommand.DEFAULT_FAILURE_CONSUMER, noReload);
+        commandInvocation.getCommandContext().printLine("Command success.");
+        if (mechanism == null) {
+            commandInvocation.getCommandContext().printLine(factorySpec.getName()
+                    + " authentication disabled for " + getSecuredEndpoint(commandInvocation.getCommandContext()));
+        } else {
+            commandInvocation.getCommandContext().printLine("Mechanism removed from " + factorySpec.getName()
```

Comment:
idem as above

Reply:
Thank-you, CLI messages are not i18n. There are no plans to localise messages.